source/common/http/async_client_impl.h

```diff
@@ -299,7 +299,6 @@ class AsyncStreamImpl : public AsyncClient::Stream,
     const Router::VirtualHost& virtualHost() const override { return virtual_host_; }
     bool autoHostRewrite() const override { return false; }
     bool includeVirtualHostRateLimits() const override { return true; }
-    const envoy::config::core::v3::Metadata& metadata() const override { return metadata_; }
     const Config::TypedMetadata& typedMetadata() const override { return typed_metadata_; }
```

Comment:
Do we want to do the same thing with typedMetadata? iiuc it comes from the same proto field

Reply:
+1

Reply:
ok, i will do it in this PR